python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -814,23 +842,18 @@ def date_range(
     if isinstance(freq, DateOffset):
         offset = freq
     elif isinstance(freq, str):
-        # Map pandas `offset alias` into cudf DateOffset `CODE`, only
-        # fixed-frequency, non-anchored offset aliases are supported.
-        mo = re.fullmatch(
-            rf'(-)*(\d*)({"|".join(_offset_alias_to_code.keys())})', freq
-        )
-        if mo is None:
+        if (
+            any(
+                x in freq.upper()
+                for x in {"Y", "A", "Q", "B", "SM", "SMS", "CBMS", "M"}
+            )
+            or "MS" in freq
```

Comment:
It looks like `L`, `ms` should be [supported aliases](https://pandas.pydata.org/docs/user_guide/timeseries.html#timeseries-offset-aliases). They're not listed in the [docstring on line 765-766](https://github.com/rapidsai/cudf/blob/cbbc061ac944aa8ff53cd2c107aa12ff8dfc644a/python/cudf/cudf/core/tools/datetimes.py#L765-L766), and it looks like `3ms` would raise an error here because it contains `M` when converted to uppercase? I feel like I'm missing something, since those values were added to `_offset_alias_to_code` in this PR.

Reply:
Yeah - the issue you pointed here is valid. The plan was to allow `ms` but reject `M` and `MS`.

Reply:
Side note: `ms` is currently not tested above because the tool we used to process datetimes from pandas does not support `ms` yet.

Reply:
I moved away from checking the unsupported ones explicitly here - because that's basically just creating the complement set of supported ones in `_from_str`. Instead, I'm wrapping `_from_str` in try-except blocks.